dev-libs/opencryptoki/opencryptoki-3.23.0.ebuild

```
@@ -0,0 +1,103 @@
+)
+
+# tests right now basically don't exist; the only available thing would
+# test against an installed copy and would kill a running pcscd, all
+# things that we're not interested to.
+RESTRICT=test
+
+src_prepare() {
+	default
+	eautoreconf
+}
+
+src_configure() {
+	# package uses ${localstatedir}/lib as the default path, so if we
+	# leave it to econf, it'll create /var/lib/lib.
+
+	# Since upstream by default seem to enable any possible token, even
+	# when they don't seem to be used, we limit ourselves to the
+	# software emulation token (swtok) and if the user enabled the tpm
+	# USE flag, tpmtok.  The rest of the tokens seem to be hardware- or
+	# software-dependent even when they build fine without their
+	# requirements, but until somebody asks for those, I'd rather not
+	# enable them.
+
+	# We don't use --enable-debug because that tinkers with the CFLAGS
+	# and we don't want that. Instead we append -DDEBUG which enables
+	# debug information.
+	use debug && append-flags -DDEBUG
+
+	local myeconfflags=(
+		--localstatedir=/var
+		--enable-fast-install
+		--enable-daemon
+		--enable-library
+		--enable-swtok
+		--disable-debug
+		--disable-icatok
+		$(use_enable tpm tpmtok)
+		--disable-ccatok
+	)
+	econf "${myeconfargs[@]}"
+}
+
+src_install() {
+	default
+
+	find "${ED}" -name '*.la' -delete || die
+
+	# Install libopencryptoki in the standard directory for libraries.
+	mv "${ED}"/usr/$(get_libdir)/opencryptoki/libopencryptoki.so* "${ED}"/usr/$(get_libdir) || die
+	rm "${ED}"/usr/$(get_libdir)/pkcs11/libopencryptoki.so || die
+	dosym ../libopencryptoki.so /usr/$(get_libdir)/pkcs11/libopencryptoki.so
+
+	# Remove compatibility symlinks as we _never_ required those and
+	# they seem unused even upstream.
+	find "${ED}" -name 'PKCS11_*' -delete || die
+
+	# We replace their ld.so and init files (mostly designed for RedHat
+	# as far as I can tell) with our own replacements.
+	rm -rf "${ED}"/etc/ld.so.conf.d "${ED}"/etc/rc.d || die
```

We tend not to use `-f` with predictable files. `-f` will silently ignore any errors.